datetime/src/main/scala/com/fortysevendeg/scalacheck/datetime/typeclasses/ScalaCheckDateTimeInfra.scala

```diff
@@ -15,4 +15,20 @@ trait ScalaCheckDateTimeInfra[D, R] {
   def addRange(dateTime: D, range: R): D
   def addMillis(dateTime: D, millis: Long): D
   def getMillis(dateTime: D): Long
+  def isBefore(dt1: D, dt2: D): Boolean
 }
+
```

Comment:
This means we have three typeclasses, one for `DateTime`, one for `Date` and one for `Time`. This feels like a smell, I'm going to have a think if this can be condensed into one class.